packages/client/src/utils/ShoppingCartContext/ShoppingCartContext.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useContext,
+  useMemo,
+  useCallback,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useUserContext } from '../../userContext';
+
+const ShoppingCartContext = createContext({});
+
+export const useShoppingCart = () => {
+  return useContext(ShoppingCartContext);
+};
+
+const ShoppingCartProvider = ({ children }) => {
+  const { user } = useUserContext();
+
+  const [pendingItems, setPendingItems] = useState(() => {
+    const items = localStorage.getItem(user?.uid);
+    const initialValue = JSON.parse(items);
+    return initialValue || [];
+  });
+
+  const deletePendingItem = useCallback(
+    (itemId) => {
+      const filteredItems = pendingItems.filter(({ id }) => id !== itemId);
+      localStorage.setItem(user?.uid, JSON.stringify(filteredItems));
+      setPendingItems(filteredItems);
+    },
+    [pendingItems, user?.uid],
+  );
+
+  const addPendingItem = useCallback(
+    (test) => {
+      const updatedPendingItems = { ...pendingItems, ...test };
+      localStorage.setItem(user?.uid, JSON.stringify(updatedPendingItems));
+      setPendingItems(updatedPendingItems);
+    },
+    [pendingItems, user?.uid],
+  );
+
+  const shoppingCartProviderValue = useMemo(
+    () => ({
+      pendingItems,
+      deletePendingItem,
+      addPendingItem,
```

Comment:
What do you mean by `pending`? The word confuses me :D You mean in the sense that they have yet to be ordered and paid for? 
I would remove that word because I think it adds more confusion than it adds clarity. The fact that they are in the cart already tells me they have not yet been ordered.

Reply:
Make sense 👍 I'll change it to orderItems soon